Bring over all the Trainings configuration and tools

stevepiercy opened the pull request:
- Begin replacing "Trainings" with "Documentation"

See checklist in #1134

stevepiercy committed:
Bring over all the Trainings configuration and tools
- Begin replacing "Trainings" with "Documentation"

stevepiercy commented:
This is a WIP. I still have a bit more work to do before it is ready for others to start using.

But if you just can't wait, check out this branch and read `docs/contributing/setup-build.md` for a quick start.

stevepiercy committed:
Use intersphinx link

stevepiercy committed:
Pin Sphinx minimum version
- Update search select menu options
- Add another Plone logo version
- Blackify

stevepiercy committed:
Add MyST for labels, clean up English grammar and syntax

stevepiercy committed:
Add the backend structure, clean up the index toctree

stevepiercy committed:
Add missing images, fix reference to static path

stevepiercy committed:
Fix copyright, author, and extra_footer values

stevepiercy committed:
Remove docs/README-doc-writers.txt as redundant

stevepiercy committed:
Remove obsolete readme

stevepiercy committed:
Add new structure

stevepiercy committed:
Clean up icons.md and whatsnew.md for proper English grammar and syntax

stevepiercy committed:
Migrate useful content from registry/index into static-resources.md

stevepiercy committed:
Migrate useful content from registry/index into static-resources.md

stevepiercy committed:
Rename files and flatten structure one level
Remove theme-package because it was added out of confusion

stevepiercy committed:
Merge and resolve differences between the two TTW files

stevepiercy committed:
Clean up recipes.md, adding labels

stevepiercy committed:
Clean up outlines of theming files
Update index for structure
Fix Sphinx warnings and errors

stevepiercy committed:
Change default option label from Trainings to Documentation

stevepiercy committed:
Fix title

stevepiercy commented:
This PR is now ready for review.

- Creates the structure shown in https://github.com/plone/documentation/issues/1137 Volto is still TBD.
- Add Makefile
- Add filter by section
- Use MyST and Sphinx Book Theme
- Add sticky TOC in navigation
- Copy code with a click
- Add Glossary
- Automatically generated index
- Add Contributing chapter

This gives us a good base from which to work.

stevepiercy committed:
pin Sphinx version to avoid badly formatted admonitions
See https://github.com/sphinx-doc/sphinx/issues/10121

stevepiercy committed:
Add genindex and remove Glossary from main page content

stevepiercy committed:
Add GitHub Actions and CONTRIBUTING.md
- Temporarily comment out some deploy steps until we have resolved
  https://github.com/plone/documentation/issues/1146

stevepiercy committed:
Quote Python version and branches

stevepiercy committed:
Remove constraints call

stevepiercy committed:
Fix linkcheck

stevepiercy committed:
Exclude linode.com from linkcheck

stevepiercy commented:
I just added GitHub actions for testing and (eventually) automatic deployment, which will come in https://github.com/plone/documentation/issues/1146

stevepiercy committed:
Add html_meta information
Add missing labels

stevepiercy committed:
Remove presentation make target

stevepiercy committed:
Remove and replace references to "trainings" in favor of "documentation"

stevepiercy committed:
Replace training with doc_section for search

stevepiercy committed:
unpin Sphinx. User error.

stevepiercy committed:
Add Netlify Python runtime version

stevepiercy committed:
Remove trailing newline. yeesh!

stevepiercy committed:
Try 3.8

stevepiercy committed:
Try 3.10 without quotes

stevepiercy committed:
Try 3.8 without quotes

stevepiercy committed:
Fix name value

stevepiercy committed:
Add Netlify logo to footer sitewide

ksuess commented on docs/conf.py:
```diff
@@ -0,0 +1,217 @@
+# a list of builtin themes.
+#
+html_theme = "sphinx_book_theme"
+
```
custom.css and print.css are not included in _build. I've build with `make html` and `gulp`.
With explicit html_static_path they are included.

```suggestion

html_static_path = ["_static"]

```

stevepiercy replied:
Ah, yes, thank you! Fixed in de0b83a5b234f7909c6c2d9883003dcad0d1055b

ksuess commented on docs/conf.py:
```diff
@@ -0,0 +1,217 @@
+html_use_opensearch = "https://docs.plone.org"
+
+html_theme_options = {
+    "google_analytics_id": "UA-1907133-6",
```
Is the Google Analytics Tracking-Code for dev different from Tracking-Code of docs.plone.org?

stevepiercy replied:
No, it is the same code.

Should I use a different one?

How do I get a different one?

ksuess replied:
To not clutter the account of live documentation a different code would be good. Or maybe removing from dev and opening a reminding issue to bring it back on relaunch is sufficient.

stevepiercy replied:
Done in 95404a22c142a3e7a5dc36c34e10755583f7df89 and https://github.com/plone/documentation/issues/1158.

ksuess reviewed (changes requested):
Add robots.txt 
with html_extra_path https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_extra_path

ksuess commented:
Ease to start contributing:

Maybe include creation of venv in `make build` :

```
python -m venv .
source bin/activate
```

and a hint to install enchant. Then make build would run without the need to read the guideline and preparing. And one bit of frustrating about the raised error is gone.

stevepiercy committed:
Add robots.txt and values for html_extra_path and html_static_path

stevepiercy commented:
> Add robots.txt with html_extra_path https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_extra_path

Fixed in de0b83a5b234f7909c6c2d9883003dcad0d1055b

stevepiercy committed:
Reorganize the build target commands
Set a relative path to `SPHINXBUILD` to avoid needing to activate the virtual environment

stevepiercy committed:
Align contributing docs with Makefile

stevepiercy commented:
> Ease to start contributing:
> 
> Maybe include creation of venv in `make build` :
> 
> ```
> python -m venv .
> source bin/activate
> ```

I moved around some things in the Makefile to get this to partially work.

Nothing I tried would activate the virtual environment and keep it activated for repeated `make html`. Instead I changed the value for `SPHINX_BUILD` to use its relative path.

> and a hint to install enchant. Then make build would run without the need to read the guideline and preparing. And one bit of frustrating about the raised error is gone.

This is in the `contributing/setup-build.md` file. Maybe we do an `if/fi` to check if enchant is in the path, else exit?

stevepiercy committed:
Revert attempt to activate the Python virtual environment in Makefile.
It breaks the build on Netlify.

stevepiercy commented:
Nope, that breaks the build on Netlify. I reverted the Makefile change and updated the instructions in `setup-build.md` in bb6c179bfff8b289251ef1d94872d7a0a4a3a94d.

stevepiercy committed:
Disable `google_analytics_id`
Either get a separate GA ID or enable this one once it is in production

stevepiercy commented:
@ksuess I've pushed some commits and responded to your review. Tag! You're it! :grin:

I was not able to figure out the virtual environment activation. I also don't think we need to do much more in the Makefile for enchant, as the instructions is the entry point for getting started.

ksuess commented:
`make build` works quite fine now. It's not even necessary to activate the venv. There's some magic that sphinx-build is found. Try 
`$ sphinx-build `
and
`$ which sphinx-build`

So you can skip the line
`source bin/activate`
in setup-build.md

stevepiercy committed:
Try using sphinx-autobuild for live preview

ksuess commented:
> `make build` works quite fine now. It's not even necessary to activate the venv. There's some magic that sphinx-build is found. Try `$ sphinx-build ` and `$ which sphinx-build`
> 
> So you can skip the line `source bin/activate` in setup-build.md

It's wrong what I wrote, I did a test from scratch. But I think it's fine now. We can come back to it later.

stevepiercy commented:
@ksuess, I added one more commit for you to test out. You'll need to run `pip install -r requirements` then `make livehtml`.

If it is OK to merge this PR as is, please let me know. I am eager to get something deployed for people to start playing with.

ksuess commented:
This can be deleted:


.PHONY: test
testlight: clean spellcheck  ## Run spellcheck

stevepiercy committed:
Remove obsolete test target
Move `all` target to the end

stevepiercy commented:
> This can be deleted:
> 
> .PHONY: test testlight: clean spellcheck ## Run spellcheck

Good catch! Fixed in 694129236df6e52e441e2fa5903d735114e9bdcc.

ksuess commented:
`make livehtml` 😻

ksuess commented:
Maybe just add a text hint with `@echo`:

.PHONY: build
build:  ## Set up training: Install requirements
	python3 -m venv . || virtualenv --clear --python=python3 .
	bin/python -m pip install --upgrade pip
	bin/pip install -r requirements.txt
	@echo
	@echo "Please activate your Python virtual environment with"
	@echo "source bin/activate"

ksuess commented:
Let's see what other reviewers find. From me a GO!

ksuess commented:
> 



> `make livehtml` 😻

I you add 
```
.PHONY: livehtml
livehtml: ## Run live updating build
```
then `make livehtml `is shown in the output  of `make`

stevepiercy committed:
Replace gulp, yarn, and npm with sphinx-autobuild

stevepiercy committed:
Add message for build
Add description comment for livehtml when calling `make`

stevepiercy commented:
I added updates per your comments, but grabbed the description from https://pypi.org/project/sphinx-autobuild/ for `make`. See 4d035adcd7389b76972b24b880c1c75ffb84dbda

ksuess commented:
As long as this PR is open: here another little one:
_source folder with txt files could be excluded via robots.txt. Maybe more than this folder…
![image](https://user-images.githubusercontent.com/1144737/150325611-89427b2d-f938-45b7-ba5f-a882c7b3d974.png)
For example https://61e8de5e50d8f50008c486f8--6-dev-docs-plone-org.netlify.app/_sources/glossary.md.txt

stevepiercy committed:
Update robots.txt

stevepiercy commented:
I updated the robots.txt, based on what shows up in `_build/html/`.

stevepiercy committed:
Use correct name for objects.inv

stevepiercy committed:
Don't both ignoring ports on localhost

stevepiercy commented:
I'm going to merge this branch after Europeans have had one more day to review it.

I want to announce that `6-dev` is ready for everyone to work on before the weekend.

polyester committed:
fix url

polyester commented on .github/workflows/build_deploy.yml:
```diff
@@ -0,0 +1,47 @@
+#        id: deploy
+#        uses: Pendect/action-rsyncer@v1.1.0
+#        env:
+#          DEPLOY_KEY: ${{secrets.DEPLOY_KEY_TRAINING}}
```
This probably needs a new key

stevepiercy replied:
Yup, that's why I commented it out. Who can provide one?

polyester replied:
@ericof should be able to do so.

stevepiercy replied:
Let's continue the conversion in https://github.com/plone/documentation/issues/1146

polyester commented on .github/workflows/build_deploy.yml:
```diff
@@ -0,0 +1,47 @@
+#          options: ''
+#          ssh_options: '-p ${{secrets.DEPLOY_PORT}}'
+#          src: '_build/html/'
+#          dest: '${{secrets.DEPLOY_USER_TRAINING}}@${{secrets.DEPLOY_SERVER_TRAINING}}:${{secrets.DEPLOY_PATH_TRAINING}}/5'
```
here also

stevepiercy committed:
Remove unnecessary phrase

stevepiercy committed:
Merge branch 'port-training-config' of github.com:stevepiercy/documentation into port-training-config

stevepiercy committed:
Add sphinx-sitemap
